sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AzureOauth2RefreshTokenGrantRequestEntityConverter.java

```diff
@@ -19,14 +19,21 @@
     @SuppressWarnings("unchecked")
     public RequestEntity<?> convert(OAuth2RefreshTokenGrantRequest refreshTokenGrantRequest) {
         RequestEntity<?> result = super.convert(refreshTokenGrantRequest);
-        Optional.ofNullable(result)
-            .map(HttpEntity::getBody)
-            .map(b -> (MultiValueMap<String, String>) b)
-            .ifPresent(body -> body.add("scope", scopeValue(refreshTokenGrantRequest)));
+        if (!isDefaultClient(refreshTokenGrantRequest)) {
+            Optional.ofNullable(result)
+                .map(HttpEntity::getBody)
+                .map(b -> (MultiValueMap<String, String>) b)
+                .ifPresent(body -> body.add("scope", scopeValue(refreshTokenGrantRequest)));
+        }
         return result;
     }
 
     private String scopeValue(OAuth2RefreshTokenGrantRequest refreshTokenGrantRequest) {
         return String.join(" ", refreshTokenGrantRequest.getClientRegistration().getScopes());
     }
+
+    private boolean isDefaultClient(OAuth2RefreshTokenGrantRequest request) {
```

Comment:
Move this method and `AZURE_CLIENT_REGISTRATION_ID` to `AzureClientRegistrationRepository`.